tests/unit/test_plugin.py

```diff
@@ -295,3 +301,72 @@ def test_local_csp_config_save():
             )
 
             assert get_csp_config(config=local_config) == test_data2
+
+
+def test_local_csp_usage_data():
+    json_data = {
+        "usage_metrics": [
+            {"usage_metric": "managed_node_count", "count": 42},
+            {"usage_metric": "monitoring", "count": 99}
+        ]
+    }
+    json_response = json.dumps(json_data, indent=2).encode('utf-8')
+    with patch(
+        'csp_billing_adapter_local.plugin.urllib.request.urlopen'
+    ) as mock_urlopen:
+        mock_urlopen.return_value.__enter__.return_value.read.return_value = \
+            json_response
+
+        with patch('csp_billing_adapter_local.plugin.urllib.request.Request'):
+            response = get_usage_data(config=local_config)
+            assert response == {'managed_node_count': 42}
```

Comment:
This indicates a config mismatch, i.e. the application reports 2 metrics we should track and report to the CSP but the test shows that we only care about 1 of those. As such in the real world we'd be ignoring 1 dimension for reporting to the CSP. Somewhere along he way such a mismatch should trigger an error, we cannot simply ignore  such a mismatch silently.

Reply:
Updated